pkg/api/router.go

```diff
@@ -104,7 +104,9 @@ func GenerateRouter(apiConf *Config, promqlConf *query.Config, client *pgclient.
 	reloadHandler := timeHandler(metrics.HTTPRequestDuration, "/-/reload", Reload(reload, apiConf.AdminAPIEnabled))
 	router.Path("/-/reload").Methods(http.MethodPost).HandlerFunc(reloadHandler)
 
-	jaeger.ExtendQueryAPIs(router, client.ReadOnlyConnection(), query)
+	if store != nil {
```

Comment:
This is added to skip creation of Jaeger store when doing promql_endpoint_integration_test.go